utils/src/main/scala/it/pagopa/interop/commons/utils/package.scala

```diff
@@ -30,4 +31,16 @@ package object utils {
     } yield (bearerToken, correlationId, ip)
   }
 
+  def withHeaders[T](
+    f: (String, String, Option[String]) => Future[T]
```

Comment:
I know this is tied to the existing `extractHeaders`, but when using this function it could be a bit tough to understand what each parameter is.
Could type aliases be an option for you here?

Reply:
It might work, but sincerely, I don't know if autocompletion of IDEs and similar will show the type alias name instead of the aliased one. I'll give it a try.